src/Avalonia.X11/NativeDialogs/LinuxStorageProvider.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Avalonia.FreeDesktop;
+using Avalonia.Platform.Storage;
+
+namespace Avalonia.X11.NativeDialogs;
+
+internal class LinuxStorageProvider : IStorageProvider
```

Comment:
Can we have "CompositeStorageProvider" instead? Something that would accept `IEnumerable<Func<Task<IStorageProvider>>` in constructor.